test/nn/test_pooling.py

```diff
@@ -376,6 +376,40 @@ def test_adaptive_pooling_zero_batch(self, dtype, device):
         mod = torch.nn.AdaptiveAvgPool3d((5, 5, 5)).to(device)
         _test_module_empty_input(self, mod, inp, check_size=False)
 
+    @onlyNativeDeviceTypes
+    @dtypes(torch.float32, torch.float64)
+    @dtypesIfCUDA(torch.float32, torch.float64, torch.bfloat16, torch.float16)
+    def test_adaptive_pooling_empty_output_size(self, dtype, device):
```

Comment:
Instead of testing the modules, what about adding error inputs to the functional OpInfos?

Error inputs may not be sufficient because they only test forward, and some of these errors may only happen when backwards is called, so those would require a new test, but I'd still suggest writing that test with the functions if possible

Reply:
@mruberry, In this PR, `adaptive_{avg, max}_pool` variants raise errors only for backward mode, but it works well for forward mode when output_size is 0. So I think error inputs are not possible in this case.

PS: I'll create another PR adding ErrorInputs for all the possible cases in these variants :)